show/reboot_cause.py

```diff
@@ -62,33 +163,34 @@ def reboot_cause(ctx):
         click.echo(reboot_cause_str)
 
 
-# 'history' subcommand ("show reboot-cause history")
+# 'all' command within 'reboot-cause'
 @reboot_cause.command()
-def history():
-    """Show history of reboot-cause"""
-    REBOOT_CAUSE_TABLE_NAME = "REBOOT_CAUSE"
-    TABLE_NAME_SEPARATOR = '|'
-    db = SonicV2Connector(host='127.0.0.1')
-    db.connect(db.STATE_DB, False)   # Make one attempt only
-    prefix = REBOOT_CAUSE_TABLE_NAME + TABLE_NAME_SEPARATOR
-    _hash = '{}{}'.format(prefix, '*')
-    table_keys = db.keys(db.STATE_DB, _hash)
-    if table_keys is not None:
-        table_keys.sort(reverse=True)
+def all():
+    if not is_smartswitch():
+        return
+    """Show cause of most recent reboot"""
+    reboot_cause_data = fetch_reboot_cause_from_db("all")
+    header = ['Device', 'Name', 'Cause', 'Time', 'User']
+    if reboot_cause_data:
+        click.echo(tabulate(reboot_cause_data, header, numalign="left"))
 
-        table = []
-        for tk in table_keys:
-            entry = db.get_all(db.STATE_DB, tk)
-            r = []
-            r.append(tk.replace(prefix, ""))
-            r.append(entry['cause'] if 'cause' in entry else "")
-            r.append(entry['time'] if 'time' in entry else "")
-            r.append(entry['user'] if 'user' in entry else "")
-            r.append(entry['comment'] if 'comment' in entry else "")
-            table.append(r)
 
-        header = ['Name', 'Cause', 'Time', 'User', 'Comment']
-        click.echo(tabulate(table, header, numalign="left"))
+# 'history' command within 'reboot-cause'
+@reboot_cause.command()
+@click.argument(
+        'module_name',
+        required=False,
+        type=click.Choice(get_all_dpu_options(), case_sensitive=False) if is_smartswitch() else None
+        )
+def history(module_name=None):
+    """Show history of reboot-cause"""
+    if not is_smartswitch() and module_name:
```

Comment:
Can we print something here instead of returning without any error/log message?

Reply:
@gpunathilell Added message to indicate that the 'module' option is supported only for smartswitch platform.